### PR TITLE
Download and unpack M2 repo cache for Kokoro builds

### PR DESCRIPTION
**Pull request by chanseokoh**
So, I've tar'ed up an M2 repo (`$HOME/.m2`) containing almost all artifacts and bundles used in Kokoro Oxygen builds. Its size is about 690MB. I uploaded it to a GCS bucket.

I tested downloading the tar and unpacking on Kokoro, and surprisingly, the whole process takes only a few seconds. No more artifact/bundle downloading (for now), so this saves a lot of time.

### CODE REVIEW

**Comment by chanseokoh** [comment]
We can update the tar when it becomes seriously out of date and Kokoro becomes downloading a lot of new artifacts. I think we can update it very infrequently. I will write a Wiki page about all these.

File: kokoro/windows/continuous.bat [hunk 1]
```diff
@@ -1,4 +1,12 @@
 @echo on
+
+rem To speed up build, download and unpack an M2 repo cache.
+pushd %USERPROFILE%
+call gsutil -q cp "gs://ct4e-m2-repositories/m2-oxygen.tar" .
```
Comment (briandealwis) [inline kokoro/windows/continuous.bat]: You may be able to use `gsutil cat xxx | tar xf -` which should be more efficient.

**Comment by chanseokoh** [comment]
The main test run of the Kokoro build in this PR took 23:03 min. (The entire Kokoro run will take ~10 more minutes.)
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 23:03 min
[INFO] Finished at: 2017-12-14T22:57:29-08:00
[INFO] Final Memory: 191M/826M
[INFO] ------------------------------------------------------------------------
```

A week ago, it took 39:00 min in #2659 ([Stackdriver log here](https://console.cloud.google.com/logs/viewer?project=cloud-tools-for-eclipse-kokoro&advancedFilter=resource.type=%22project%22%0Aresource.labels.project_id=%22cloud-tools-for-eclipse-kokoro%22%0AlogName=%22projects/cloud-tools-for-eclipse-kokoro/logs/kokoro-google-cloud-eclipse-gcp_windows-presubmit-955-20171208-124559%22)):
```
[INFO] ------------------------------------------------------------------------ 
[INFO] BUILD SUCCESS 
[INFO] ------------------------------------------------------------------------ 
[INFO] Total time: 39:00 min 
[INFO] Finished at: 2017-12-08T13:29:23-08:00 
[INFO] Final Memory: 236M/1689M 
[INFO] ------------------------------------------------------------------------ 
```

File: kokoro/windows/continuous.bat (hunk 1 above)
Comment (chanseokoh) [reply kokoro/windows/continuous.bat]: Nice.

**Pull request merged**